### PR TITLE
Issues #17, #18, #20, #25 - Indoor map

### PR DESCRIPTION
**Pull request by NicholasNagy**
#17, #18, #20, #25   

In this pull request, I aim to bring in the following functionality:

- The ability to navigate between 2 rooms on the 8th floor of the Hall building (no other buildings are currently supported

- The ability to generate the shortest path between 2 rooms on the same floor.

- Displaying room numbers, and information on the indoor map

### CODE REVIEW

**Comment by NicholasNagy** [comment]
> Wow, amazing work! 😍
> I ran the code and tried various points on the map. I found some issues:
> 
> * Although these are rooms, the map displays a path between the two escalators
>   From: hall-806.03
>   To: hall-840
> * The path between these two rooms is generating over the glass walls next to the escalators
>   From: hall-820.01
>   To: hall-840
> 
> I also ran all UI and unit tests, everything is passing 🙂
> 
> Please look into the two navigation scenarios and commenting your code 😬

I made a fix to the SVG file, so now no paths will be generated that cross through the glass wall

File: app/src/main/java/com/droidhats/campuscompass/models/Location.kt [hunk 1]
```diff
@@ -15,19 +15,24 @@ import kotlinx.android.parcel.Parcelize
 */
 abstract class Location {
     abstract val name: String
-    abstract val coordinate: LatLng
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/models/Location.kt]: Great decision!

File: app/src/main/java/com/droidhats/campuscompass/views/SearchFragment.kt [hunk 2]
```diff
@@ -298,7 +310,7 @@ class SearchFragment : Fragment()  {
                     selectedTransportationMethod = NavigationRoute.TransportationMethods.BICYCLE.string
                 }
                 R.id.radio_transport_mode_shuttle -> {
-                    selectedTransportationMethod = NavigationRoute.TransportationMethods.SHUTTLE.string
+                     selectedTransportationMethod = NavigationRoute.TransportationMethods.SHUTTLE.string
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/SearchFragment.kt]: Not a fan of this extra space

File: app/src/main/java/com/droidhats/campuscompass/repositories/IndoorLocationRepository.kt [hunk 3]
```diff
@@ -16,11 +21,72 @@ class IndoorLocationRepository private constructor(private val indoorLocationDao
...
+                }
+                val newClass = IndoorLocation(
+                    classRoom.getID(),
+                    convertIDToName(classRoom.getID(), building.getIndoorInfo().first, floorNumber),
+                    floorNumber,
+                    "classroom",
+                    building.coordinate.latitude,
+                    building.coordinate.longitude
+                )
+                println(newClass)
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/repositories/IndoorLocationRepository.kt]: Please remove this.

File: app/src/main/java/com/droidhats/campuscompass/views/FloorFragment.kt [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProviders
+import androidx.navigation.fragment.findNavController
+import com.droidhats.mapprocessor.ProcessMap
+import com.caverock.androidsvg.SVG
+import com.droidhats.campuscompass.R
+import com.droidhats.campuscompass.viewmodels.FloorViewModel
+import com.mancj.materialsearchbar.MaterialSearchBar
+import com.otaliastudios.zoom.ZoomImageView
+import kotlinx.android.synthetic.main.search_bar_layout.*
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/FloorFragment.kt]: Please don't use * when importing packages.
We have discussed it's better to specify the required components.

File: app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt [hunk 5]
```diff
@@ -26,14 +26,12 @@ import androidx.fragment.app.Fragment
...
+import androidx.navigation.fragment.navArgs
 import com.droidhats.campuscompass.MainActivity
 import com.droidhats.campuscompass.R
 import com.droidhats.campuscompass.adapters.SearchAdapter
 import com.droidhats.campuscompass.helpers.Subject
-import com.droidhats.campuscompass.models.Building
-import com.droidhats.campuscompass.models.CalendarEvent
-import com.droidhats.campuscompass.models.GooglePlace
-import com.droidhats.campuscompass.models.NavigationRoute
+import com.droidhats.campuscompass.models.*
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt]: Please don't use * when importing packages.
We have discussed it's better to specify the required components.

File: app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt [hunk 6]
```diff
@@ -47,13 +45,7 @@ import com.google.android.gms.maps.GoogleMap
...
 import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
-import com.google.android.gms.maps.model.Marker
-import com.google.android.gms.maps.model.Polygon
-import com.google.android.gms.maps.model.MarkerOptions
-import com.google.android.gms.maps.model.LatLng
-import com.google.android.gms.maps.model.PolylineOptions
-import com.google.android.gms.maps.model.Polyline
-import com.google.android.gms.maps.model.MapStyleOptions
+import com.google.android.gms.maps.model.*
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt]: As specified before, please don't use * when importing packages.

**Pull request merged**